src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -132,9 +133,22 @@ internal ImmutableArray<FunctionMetadata> LoadFunctionMetadata(bool forceRefresh
             ImmutableArray<FunctionMetadata> immutableFunctionMetadata;
             var workerConfigs = _languageWorkerOptions.CurrentValue.WorkerConfigs;
 
-            IFunctionMetadataProvider metadataProvider = new AggregateFunctionMetadataProvider(_loggerFactory.CreateLogger<AggregateFunctionMetadataProvider>(), dispatcher, _functionMetadataProvider, _scriptOptions);
+            bool workerIndexing = Utility.CanWorkerIndex(workerConfigs, SystemEnvironment.Instance);
+            IFunctionMetadataProvider functionMetadataProvider = null;
 
-            immutableFunctionMetadata = metadataProvider.GetFunctionMetadataAsync(workerConfigs, SystemEnvironment.Instance, forceRefresh).GetAwaiter().GetResult();
+            if (workerIndexing)
+            {
+                functionMetadataProvider = _functionMetadataProviders.Where(p => p.GetType() == typeof(WorkerFunctionMetadataProvider)).FirstOrDefault();
+                immutableFunctionMetadata = functionMetadataProvider.GetFunctionMetadataAsync(workerConfigs, SystemEnvironment.Instance, forceRefresh).GetAwaiter().GetResult();
```

Comment:
This will only work when specialization has happened before this step, so we have Customer payload available at this point. Is this correct? How can we make sure that it is happening in code?

Reply:
It will actually work before or after specialization. The only difference is that `workerIndexing` will only return true after specialization. ( `false` once we flip the feature flag). It ends up getting called at both times, before and after specialization. We clear the cache after specialization, which gives us actual customer payload.